packages/beacon-node/src/chain/blocks/verifyBlocksSignatures.ts

```diff
@@ -46,17 +46,20 @@ export async function verifyBlocksSignatures(
     }
   }
 
-  if (blocks.length === 1 && opts.seenTimestampSec !== undefined) {
-    const recvToSigVer = Date.now() / 1000 - opts.seenTimestampSec;
-    metrics?.gossipBlock.receivedToSignaturesVerification.observe(recvToSigVer);
-    logger.verbose("Verified block signatures", {slot: blocks[0].message.slot, recvToSigVer});
-  }
-
   // `rejectFirstInvalidResolveAllValid()` returns on isValid result with its index
   const res = await rejectFirstInvalidResolveAllValid(isValidPromises);
   if (!res.allValid) {
     throw new BlockError(blocks[res.index], {code: BlockErrorCode.INVALID_SIGNATURE, state: preState0});
   }
+
+  const verifySignaturesTime = Date.now();
+  if (blocks.length === 1 && opts.seenTimestampSec !== undefined) {
+    const recvToSigVer = verifySignaturesTime / 1000 - opts.seenTimestampSec;
+    metrics?.gossipBlock.receivedToSignaturesVerification.observe(recvToSigVer);
+    logger.verbose("Verified block signatures", {slot: blocks[0].message.slot, recvToSigVer});
+  }
```

Comment:
What's the reason to re-order this @g11tech ?

CC: @tuyennhv

Reply:
umm i am assuming the aim was to clock all the time including `rejectFirstInvalidResolveAllValid`, lemme know @tuyennhv what is the correct expectation here

Reply:
this is correct 👍